edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Core/PersistedTokenCredentialsCache.cs

```diff
@@ -35,34 +37,59 @@ public async Task Add(IClientCredentials clientCredentials)
 
         public async Task<Option<IClientCredentials>> Get(IIdentity identity)
         {
-            Option<string> tokenOption = await this.encryptedStore.Get(identity.Id);
-            return tokenOption.Map(token =>
+            Option<string> tokenCredentialsDataOption = await this.encryptedStore.Get(identity.Id);
+            return tokenCredentialsDataOption.FlatMap(t =>
             {
                 try
                 {
-                    Events.Retrieved(identity.Id);
-                    return Option.Some(new TokenCredentials(identity, token, string.Empty) as IClientCredentials);
+                    var tokenCredentialsData = t.FromJson<TokenCredentialsData>();
+                    Events.Retrieved(identity.Id);                    
+                    return Option.Some(new TokenCredentials(identity, tokenCredentialsData.Token, string.Empty, tokenCredentialsData.IsUpdatable) as IClientCredentials);
                 }
                 catch (Exception e)
```

Comment:
nit: extract this in a method that tries to deserialize the json and return none if it fails. So the code from the current catch clause is executed when none was returned

Reply:
Done